Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs

```diff
@@ -231,38 +240,26 @@ public virtual void OnSourceDetected(SourceStateEventData eventData)
         /// <inheritdoc />
         public virtual void OnSourceLost(SourceStateEventData eventData)
         {
-            if (eventData.Controller != null)
+            if (Pointer != null && eventData.Controller != null)
             {
                 for (int i = 0; i < eventData.InputSource.Pointers.Length; i++)
                 {
                     // If a source is lost that's using this cursor's pointer, we decrement the count to set the cursor state properly.
                     if (eventData.InputSource.Pointers[i].PointerId == Pointer.PointerId)
                     {
-                        var basePointer = eventData.InputSource.Pointers[i] as BaseControllerPointer;
-
-                        if (basePointer != null &&
-                            basePointer.DestroyOnSourceLost)
-                        {
-                            SourceDownIds.Remove(eventData.SourceId);
-                            Destroy(gameObject);
-                            return;
-                        }
-
                         visibleSourcesCount--;
+                        break;
                     }
                 }
             }
 
-            if (!IsSourceDetected)
-            {
                 SourceDownIds.Remove(eventData.SourceId);
 
-                if (SetVisibilityOnSourceDetected)
+            if (!IsSourceDetected && SetVisibilityOnSourceDetected)
                 {
```

Comment:
Not sure if this is just GitHub formatting, but the indentation here seems inconsistent with the line above.

Reply:
yeah, the indentation is wrong (probably a merge artifact) - I'll fix it

Reply:
well, it looks good locally now, but still looks bad here :/ hopefully it is just GitHub formatting at this point.